lib/util/propTypes.js

```diff
@@ -172,6 +174,7 @@ module.exports = function propTypesInstructions(context, components, utils) {
     UnionTypeAnnotation: function(annotation, parentName, seen) {
       const unionTypeDefinition = {
         type: 'union',
+        /** @type {any} */
```

Comment:
lets avoid any use of any :-)

Reply:
Tsc does not accept `any[]` and `any[]|true`.

Reply:
`unknown[]`? `unknown[] | true`? you can certainly do any of those in TS, altho maybe not in jsdoc.

Reply:
Tsc does not accept `unknown[]` `unknown[]|true` also.

Reply:
just in jsdoc? it certainly works in TS.

Reply:
`unionTypeDefinition.children` is at a time assigned `true`, so it cannot be `unknown[]`. `unionTypeDefinition.children.length` is used, so it cannot be `unknown[]|true`.

Reply:
then that implies there's a bug, or else type narrowing would have removed `true` prior to the `.length` access?